src/org/isf/medicalstock/service/MovementIoOperationRepositoryImpl.java

```diff
@@ -214,23 +226,24 @@ private String _getMovementForPrint(
 		boolean paramQuery = false;
 		
 		
-		query = "select MMV_ID from ((MEDICALDSRSTOCKMOVTYPE join MEDICALDSRSTOCKMOV on MMVT_ID_A = MMV_MMVT_ID_A) " +
-				"join (MEDICALDSR join MEDICALDSRTYPE on MDSR_MDSRT_ID_A=MDSRT_ID_A) on MMV_MDSR_ID=MDSR_ID) " +
-				"left join WARD on MMV_WRD_ID_A=WRD_ID_A " +
-				"left join MEDICALDSRLOT on MMV_LT_ID_A=LT_ID_A " +
-				"LEFT JOIN SUPPLIER ON MMV_FROM = SUP_ID " +
+		query = "select mov.code from Movement mov " +
+				"join mov.type movtype " +
+				"join mov.medical med " +
+				"left join mov.ward ward " +
+				"left join mov.lot lot " +
+				"left join mov.supplier sup " +
 				"where ";
 
 		if ((medicalDescription != null) || (medicalTypeCode != null)) 
 		{
 			if (medicalDescription == null) 
 			{
-				query += "(MDSR_MDSRT_ID_A = \"" + medicalTypeCode + "\") ";
+				query += "(medtype.code=:medicalType) ";
```

Comment:
JPA has something called criteria builder which fits perfect form building queries with optional where clauses like this. It should make code much cleaner and safer. I see that there are named parameters but building queries manually can easily lead to mistake and sql injection vulnerability

https://vladmihalcea.com/query-entity-type-jpa-criteria-api/

Reply:
this is awesome! I already changed one method to use the criteria api and it is amazing how much cleaner the code is now! I will push after replacing the other methods using this Query-String-Build-approach as well.